policy/policy-authority.go

```diff
@@ -17,23 +18,31 @@ import (
 // PolicyAuthorityImpl enforces CA policy decisions.
 type PolicyAuthorityImpl struct {
 	log *blog.AuditLogger
+	Db  *PolicyAuthorityDatabaseImpl
 
+	EnforceWhitelist bool
 	PublicSuffixList map[string]bool // A copy of the DNS root zone
-	Blacklist        map[string]bool // A blacklist of denied names
 }
 
 // NewPolicyAuthorityImpl constructs a Policy Authority.
-func NewPolicyAuthorityImpl() *PolicyAuthorityImpl {
+func NewPolicyAuthorityImpl(common cmd.CommonConfig) (*PolicyAuthorityImpl, error) {
```

Comment:
Please put driver, connect, and enforceWhiteList as arguments to this instead of passing in the config.

Inject a created Impl into the functions that need it instead of carrying this config object around